Engine/EditorUI/Panel/PanelStateMachine.cpp

```diff
@@ -34,6 +41,18 @@ PanelStateMachine::~PanelStateMachine()
 
 void PanelStateMachine::Render()
 {
+	modified_by_user = false;
+	//initialize GO to get corresponding animation controller
+	if (App->editor->selected_game_object != nullptr)
+	{
+		for (auto & component : App->editor->selected_game_object->components)
```

Comment:
If you call the GetComponent method of the gameobject this for loop is already done there, you can check it here https://github.com/OnionGalaxy/LittleOrionEngine/blob/develop/Engine/Main/GameObject.cpp#L514. You need just to call `animation_comp = (ComponentAnimation*)App->editor->selected_game_object->GetComponent(Component::ComponentType::ANIMATION); `here